File: backend/LessonRunner.Domain/Courses/Course.cs
using LessonRunner.Domain.Common;

namespace LessonRunner.Domain.Courses;

public sealed class Course : Entity
{
    public required string Name { get; set; }
    public required string Subject { get; set; }
    public required string Level { get; set; }
    public string Description { get; set; } = string.Empty;
    public List<CourseLesson> Lessons { get; set; } = [];
    public DateTimeOffset CreatedAt { get; init; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: frontend/lesson-runner-web/src/types/course.ts
export type CourseLesson = {
  lessonId: string;
  lessonTitle: string;
  subject: string;
  level: string;
  order: number;
};

export type CourseSummary = {
  id: string;
  name: string;
  subject: string;
  level: string;
  description: string;
  lessonCount: number;
};

export type CourseDetails = CourseSummary & {
  lessons: CourseLesson[];
};

export type UpsertCourseRequest = {
  name: string;
  subject: string;
  level: string;
  description: string;
  lessonIds: string[];
};
